ambersim/logger/logger.py

```diff
@@ -0,0 +1,146 @@
+import jax
+import wandb
+from torch.utils.tensorboard import SummaryWriter
+
+
+class BaseLogger:
+    """Base logger interface that defines common methods for logging metrics and parameters.
+
+    Attributes:
+        log_dir (str): Directory where logs are stored. If None, default locations are used.
+    """
+
+    def __init__(self, log_dir=None):
```

Comment:
```suggestion
    def __init__(self, log_dir: Union[str, Path]=None):
```
Throughout this PR, add typing. Don't forget to `from typing import Union` and `from pathlib import Path` for this specific suggestion - I'll ignore the rest of the missing types.